monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx

```diff
@@ -261,25 +264,31 @@ const InstalledPlugins = (props) => {
 
   return (
     <Stack spacing={2} height='100%'>
-      <Grid container spacing={2}>
-        <Grid xs={4} item
-              sx={{alignItems: 'flex-end', display: 'flex'}}>
-          <SearchBar setQuery={onSearchChanged} />
-        </Grid>
-        <Grid xs={3} item>
-          <TypeFilter allPlugins={installedPlugins}
-                        filters={filters}
-                        setFilters={setFilters}
-                        className={'type-filter-box'}/>
-        </Grid>
-        <Grid xs={2} item />
-        <Grid xs={3} item >
-          <MonkeyToggle options={[{value: 'all', label: 'All'},{value: 'upgradable', label: 'Upgradable'}]}
-                      setSelectedValues={onToggleChanged}/>
-        </Grid>
-      </Grid>
+      {installedPlugins?.length > 0 && (
+        <>
+          <Grid container spacing={2}>
+            <Grid xs={4} item
+                  sx={{alignItems: 'flex-end', display: 'flex'}}>
+              <SearchBar setQuery={onSearchChanged}/>
+            </Grid>
+            <Grid xs={3} item>
+              <TypeFilter allPlugins={installedPlugins}
+                          filters={filters}
+                          setFilters={setFilters}
+                          className="type-filter-box"/>
+            </Grid>
+            <Grid xs={2} item/>
+            <Grid xs={3} item>
+              <MonkeyToggle options={[{value: 'all', label: 'All'}, {value: 'upgradable', label: 'Upgradable'}]}
+                            setSelectedValues={onToggleChanged}/>
+            </Grid>
+          </Grid>
+        </>
+      )
+      }
       <PluginTable plugins={displayedPlugins}
                    loadingMessage="Loading all available plugins..."
+                   noRowsOverlayMessage={refreshInstalledPluginsFailure ? FETCHING_ERROR_MESSAGE : (installedPlugins?.length === 0 && NO_INSTALLED_PLUGINS_MESSAGE)}
```

Comment:
```suggestion
                   noRowsOverlayMessage={refreshInstalledPluginsFailure ? FETCHING_ERROR_MESSAGE :  NO_INSTALLED_PLUGINS_MESSAGE}
```

Reply:
This is an intended behavior. If the are plugins but the filters won't return any results you'll see 'no rows'. If there aren't plugins at all (empty list), you'll see NO_INSTALLED_PLUGINS_MESSAGE.

Reply:
Ahh, ok. Let's extract these to a method to make sure these 3 outcomes are easier to follow

Reply:
same for available plugins